### PR TITLE
Wip aj443 bounds info api refactor

### PR DESCRIPTION
**Pull request by gameboo**

### CODE REVIEW

**Comment by PeterRugg** [comment]
It would be good to run TestRIG between Flute with and without the changes to do a rough check of correctness (especially the capDecode test), since there's quite a lot to look through here. Also, checking Flute meets timing would be good (maybe using gfe-synth-experimental with a Flute pointing at this commit).

**Comment by ivanmgribeiro** [comment]
I've run this for 1000 x 4 sets of tests using TestRIG and Flute, so it _should_ work properly. I haven't checked timing.
I also have a question: do you think it would be any cheaper (in terms of hardware block usage) to getBoundsInfo on a CapMem rather than going CapMem -> CapPipe and then getBoundsInfo?

**Comment by gameboo** [comment]
Hi there. I've just approved #10, but it looks like that PR will need to do some conflict resolution to merge. This one is not currently conflicting. May well not happen at all as I don't remember off the top of my head what got touched here, but to avoid tempting the devil here and resolve conflicts once for #10 and once for #9, if people are happy with this PR, could we click merge while #9 has no conflicts?
@PeterRugg @jonwoodruff @ivanmgribeiro @francislaus

File: CHERICC_Fat.bsv [hunk 1]
```diff
@@ -836,10 +964,13 @@ function MetaInfo getMetaInfo (CapFat cap);
     , baseCorrection : baseCorrection };
 endfunction
 
-// ===============================================================================
-// Typeclass instance for interface
-
-typedef Bit#(TAdd#(1, CapW)) CapMem;
+// XXX TODO
```
Comment (francislaus) [inline CHERICC_Fat.bsv:967]: Is it still a TODO?

**Pull request merged**